code/ui/blocks/src/blocks/useOf.ts

```diff
@@ -0,0 +1,35 @@
+import type {
+  DocsContextProps,
+  NormalizedProjectAnnotations,
+  ResolvedModuleExport,
+} from '@storybook/types';
+import { useContext } from 'react';
+import { DocsContext } from './DocsContext';
+
+export type Of = Parameters<DocsContextProps['resolveModuleExport']>[0];
+
+/**
+ * A hook to resolve the `of` prop passed to a block.
+ * will return the resolved module as well as project annotations
+ * project annotations are handy because if the resolved module is a meta,
+ * it will be the meta as-is from the CSF file, and it won't inherit the global annotations.
```

Comment:
Maybe we just do the combination inside `useOf`?

Reply:
It certainly would be a better user-facing API.

I think we should only do that if we can do it in a way that is consistent with stories, so I guess we should use [`prepareStory`](https://github.com/storybookjs/storybook/blob/b8ff9f251a15a222ae460f6a29db240ca4d94870/code/lib/preview-api/src/modules/store/csf/prepareStory.ts) for that, with an empty object for story annotations. (we don't need to do this when `of = story`, and we can't do it when `of = component`).

but the typings for `prepareStory` don't really allow for empty storyAnnotations - we at least need a story ID. So I'm unsure what to do here.

Reply:
Discussed and we can do this separately,